### DISCUSSION TITLE
Should ObjectMapper throw IOException?

### DISCUSSION THREAD

**Discussion by mikehearn** [micronaut-projects/micronaut-serialization#946] [Ideas]
### Issue description

I was a bit surprised to discover that `ObjectMapper.writeValueAsString` throws `IOException` even though it's not doing IO. There is no description of when this can be thrown or why. Perhaps there should be a tighter / clearer exception type introduced in Micronaut 5?

**Comment by yawkat** [comment]
The method can still throw a SerdeException (which extends IOException), and any IOException that the encoder throws (format dependent, e.g. field size limits), so I don't think it can be more narrow.
